mathics/builtin/datentime.py

```diff
@@ -126,62 +127,63 @@ def apply(self, evaluation):
             return SymbolInfinity
 
 
-class TimeConstrained(Builtin):
-    """
-    <dl>
-    <dt>'TimeConstrained[$expr$, $t$]'
-        <dd>'evaluates $expr$, stopping after $t$ seconds.'
-    <dt>'TimeConstrained[$expr$, $t$, $failexpr$]'
-        <dd>'returns $failexpr$ if the time constraint is not met.'
-    </dl>
-    >> TimeConstrained[Integrate[Sin[x]^1000000,x],1]
-    = $Aborted
-
-    >> TimeConstrained[Integrate[Sin[x]^1000000,x], 1, Integrate[Cos[x],x]]
-    = Sin[x]
-
-    >> s=TimeConstrained[Integrate[Sin[x] ^ 3, x], a]
-     : Number of seconds a is not a positive machine-sized number or Infinity.
-     = TimeConstrained[Integrate[Sin[x] ^ 3, x], a]
-
-    >> a=1; s
-    = -Cos[x] + Cos[x] ^ 3 / 3
-
-    Possible issues: for certain time-consuming functions (like simplify)
-    which are based on sympy or other libraries, it is possible that
-    the evaluation continues after the timeout. However, at the end of the evaluation, the function will return $\\$Aborted$ and the results will not affect
-    the state of the mathics kernel.
-
-    """
-
-    attributes = ('HoldAll',)
-    messages = {
-        'timc': 'Number of seconds `1` is not a positive machine-sized number or Infinity.',
-    }
-
-    def apply_2(self, expr, t, evaluation):
-        'TimeConstrained[expr_, t_]'
-        return self.apply_3(expr, t, SymbolAborted, evaluation)
-
-    def apply_3(self, expr, t, failexpr, evaluation):
-        'TimeConstrained[expr_, t_, failexpr_]'
-        t = t.evaluate(evaluation)
-        if not t.is_numeric():
-            evaluation.message('TimeConstrained', 'timc', t)
-            return
-        try:
-            t = float(t.to_python())
-            evaluation.timeout_queue.append((t, datetime.now().timestamp()))
-            request = lambda : expr.evaluate(evaluation)
-            res = run_with_timeout_and_stack(request, t, evaluation)
-        except TimeoutInterrupt:
-            evaluation.timeout_queue.pop()
-            return failexpr.evaluate(evaluation)
-        except:
+if sys.platform != "win32":
```

Comment:
Is  TimeConstrained what is failing? or is the expression we have chosen to test it?

Reply:
I believe the implementation is Unix/POSIX specific. I haven't investigated though.